mypy/plugins/attrs.py

```diff
@@ -803,7 +803,7 @@ def _make_frozen(ctx: mypy.plugin.ClassDefContext, attributes: list[Attribute])
         else:
             # This variable belongs to a super class so create new Var so we
             # can modify it.
-            var = Var(attribute.name, ctx.cls.info[attribute.name].type)
+            var = Var(attribute.name, attribute.init_type)
```

Comment:
Actually, this is problematic with converters, e.g.
```python
def converter(s: str) -> int:
    return int(s)

@attrs.define
class C:
    x: int = attrs.field(converter=converter)
```

The `attribute.init_type` will be `str` here :(

Reply:
Ahh, nm, by the time it's deserialized, it's `int` again (there's also converter_init_type).